Validate hostname protocol before saving

msfjarvis opened the pull request:
## :loudspeaker: Type of change
<!--- Put an `x` in the boxes that apply -->
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring


## :scroll: Description
Extremely crude valiation of the protocol section of the given hostname. This was hastily thrown together and ideally the final result should look nothing like this.


## :bulb: Motivation and Context
Entering `https://github.com` in hostname will currently set the remote to `https://https://github.com` and break operations, which is rather undesirable.

## :green_heart: How did you test it?


## :pencil: Checklist
<!--- Put an `x` in the boxes that apply -->
- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code


## :crystal_ball: Next steps
Possibly validate Ssh path similarly.

## :camera_flash: Screenshots / GIFs
<!--- Mandatory for UI changes -->
<!-- Uncomment the next line and replace it with links to your screenshots. -->
<!--
<img src="https://placekitten.com/260/260" width="260">&emsp;<img src="https://placekitten.com/300/300" width="260">
-->

FabianHenneke commented:
I refactored this a little, caught the relevant exception and added support for SSH. Good catch, this would have definitely lead to a bug report!

msfjarvis commented:
> I refactored this a little, caught the relevant exception and added support for SSH. 

:+1: 

>Good catch, this would have definitely lead to a bug report!

It already did, just on Telegram rather than GitHub :P

msfjarvis commented:
Rebased on master

FabianHenneke commented on app/src/main/java/com/zeapo/pwdstore/git/BaseGitActivity.kt:
```diff
@@ -97,12 +99,31 @@ abstract class BaseGitActivity : AppCompatActivity() {
+                val parsedUrl = try {
+                    URL(urlWithFreeEntryScheme)
+                } catch (_: MalformedURLException) {
+                   return false
```
```suggestion
                    return false
```

FabianHenneke commented:
Force-pushed the lint corrections.
